challenges/cloud/challenge.md

```diff
@@ -0,0 +1,379 @@
+Start with a brief design document that covers the edge cases and design approach. At
+Teleport, we prefer Markdown for
+[our designs](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md).
+
+Be sure to cover the following in your design:
+
+* Developer workflow
+* Level 3+: Build, Release
+* Level 4+: State management and Delivery
+* Level 5+: Reconciliation, Conflicts, and Automation
+
+Open a pull request with your design and share a link with the reviewers via Slack.
+After the document is approved by at least 2 reviews, move on to the implementation.
+
+A few notes about the design document:
+
+* We expect the design document to be completed roughly within the first week.
+  This is to ensure you have enough time to work on the implementation.
+* Avoid writing an overly detailed design document. Two to three pages is
+  sufficient.
+* Avoid sending us draft design documents, it is difficult to evaluate what
+  parts are draft and which parts are complete. Instead we encourage asking
+  questions in Slack and sharing a design document that is ready to be
+  reviewed.
+
+## Server & Automation
+
+Once the design document is approved, begin working on adding features to the
+server implementation that includes automation. A key feature of this challenge 
+is to produce a self-contained GitHub project that automates away as much as 
+possible. This should include code quality, testing, build, and deployment.
+
+Expect the interview panel to clone the repository and execute one or more 
+Make targets that build and test a working solution. 
+
+Add a couple of high quality tests that cover happy and unhappy scenarios. Keep
+external dependencies as low or clear as possible. Expect the reviewers to
+checkout the repository and execute the tests themselves.
+
+Do not try to achieve full test coverage. This will take too long. Write enough
+to exercise the different key components and show they are working as
+intended, as well as to demonstrate your approach to automation.
+
+## Deployment
+
+Levels 3+ include a deployment step which should deploy the solution to a local
+Kubernetes cluster. The choice of which Kubernetes cluster is up to you, but 
+please choose one that has the ability to run on macOS and Linux.
+
+## Key Dependencies
+
+You are welcome to solve this challenge using tools you are most familiar with,
+but, at a minimum, please include the following:
+
+* Go
+* make
+* Docker
+
+## Level 1
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+
+## Level 2
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of the Kubernetes Deployment
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute tests against a locally configured Kubernetes config
```

Comment:
I wasn't sure what this meant.

Reply:
It's certainly somewhat vague (on purpose?) but the intent was to execute some level of testing against a locally running kubernetes cluster, for instance:
 - deploying the server in k8s and a deployment crd
 - Verifying the server can find the deployment
 - Verifying the server can report the correct number of replicas

Something testing, but closer to integration testing needing to deal with a few more moving parts than a simple mocked unit test